src/sentinel.rs

```diff
@@ -218,8 +223,8 @@ impl<'a, Request, Name>
             .map(|(name, public_keys, _)| {
                 (name, public_keys.into_iter()
                                   .filter(|&(_, ref count)| *count >= self.keys_threshold)
-                                  .map(|(public_key, _)| public_key)
-                                  .collect::<Vec<PublicKey>>())})
+                                  .map(|(public_key, _)| PublicKey(public_key))
+                                  .collect::<_>())})
```

Comment:
I'm not black or white with this one, I would say if it's obvious from the context and the context is small than it's all right to use _. I can change it back if you insist.